providers/implementations/signature/dsa.c

```diff
@@ -471,20 +471,24 @@ static int dsa_set_ctx_params(void *vpdsactx, const OSSL_PARAM params[])
     return 1;
 }
 
-static const OSSL_PARAM known_settable_ctx_params[] = {
+static const OSSL_PARAM settable_ctx_params[] = {
     OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_DIGEST, NULL, 0),
     OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_PROPERTIES, NULL, 0),
     OSSL_PARAM_END
 };
 
+static const OSSL_PARAM settable_ctx_params_no_digest[] = {
+    OSSL_PARAM_END
+};
+
 static const OSSL_PARAM *dsa_settable_ctx_params(void *vpdsactx,
                                                  ossl_unused void *provctx)
 {
     PROV_DSA_CTX *pdsactx = (PROV_DSA_CTX *)vpdsactx;
 
     if (pdsactx != NULL && !pdsactx->flag_allow_md)
-        return &known_settable_ctx_params[2];
-    return known_settable_ctx_params;
+        return settable_ctx_params_no_digest;
```

Comment:
I wonder what happens if NULL is returned instead...

Reply:
IMO that would be wrong as that would indicate an error to the application. Not just empty list.